src/sql/workbench/services/restore/browser/restoreDialog.ts

```diff
@@ -634,9 +634,12 @@ export class RestoreDialog extends Modal {
 	}
 
 	private onSourceDatabaseChanged(selectedDatabase: string) {
-		this.viewModel.sourceDatabaseName = selectedDatabase;
-		this.viewModel.selectedBackupSets = undefined;
-		this.validateRestore(true);
+		// This check is to avoid any unnecessary even firing (to remove flickering)
```

Comment:
I'd make this comment more generic and just be something like `Don't update model if the source database didn't change`